libs/utils/android/workloads/uibench.py

```diff
@@ -0,0 +1,143 @@
+        logging.debug('%14s - Workload created', 'UiBench')
+
+    def run(self, exp_dir, test_name, duration_s, collect=''):
+        activity = '.' + test_name + 'Activity'
+
+        # Initialize energy meter results
+        nrg_report = None
+
+        # Press Back button to be sure we run the video from the start
+        System.menu(self.target)
+        System.back(self.target)
+
+        # Close and clear application
+        System.force_stop(self.target, self.package, clear=True)
+
+        # Set airplane mode
+        System.set_airplane_mode(self.target, on=True)
+
+        # Start the main view of the app which must be running
+        # to reset the frame statistics.
+        System.monkey(self.target, self.package)
+
+        # Force screen in PORTRAIT mode
+        Screen.set_orientation(self.target, portrait=True)
+
+        # Reset frame statistics
+        System.gfxinfo_reset(self.target, self.package)
+        sleep(1)
+
+        # Clear logcat
+        os.system(self._adb('logcat -c'));
+
+        # Regexps for benchmark synchronization
+        start_logline = r'ActivityManager: START.*'\
+                         'cmp=com.android.test.uibench/{}'.format(activity)
+        UIBENCH_BENCHMARK_START_RE = re.compile(start_logline)
+        logging.debug("START string [%s]", start_logline)
+
+        # Parse logcat output lines
+        logcat_cmd = self._adb(
+                'logcat ActivityManager:* System.out:I *:S BENCH:*'\
+                .format(self.target.adb_name))
+        self.logger.info("%s", logcat_cmd)
+
+        # Start the activity
+        System.start_activity(self.target, self.package, activity)
+        logcat = Popen(logcat_cmd, shell=True, stdout=PIPE)
+        while True:
+
+            # read next logcat line (up to max 1024 chars)
+            message = logcat.stdout.readline(1024)
+
+            # Benchmark start trigger
+            match = UIBENCH_BENCHMARK_START_RE.search(message)
+            if match:
+                if 'energy' in collect and self.te.emeter:
+                    self.te.emeter.reset()
+                self.logger.debug("Benchmark started!")
+                break
+
+        # Run the workload for the required time
+        logging.info('Benchmark [%s] started, waiting %d [s]',
+                     activity, duration_s)
+        sleep(duration_s)
+        self.logger.debug("Benchmark done!")
+
+        if 'energy' in collect and self.te.emeter:
+            nrg_report = self.te.emeter.report(exp_dir)
+
+        # Get frame stats
+        db_file = os.path.join(exp_dir, "framestats.txt")
+        System.gfxinfo_get(self.target, self.package, db_file)
+
+        # Close and clear application
+        System.force_stop(self.target, self.package, clear=True)
+
+        # Go back to home screen
+        System.home(self.target)
+
+        # Switch back to screen auto rotation
+        Screen.set_orientation(self.target, auto=True)
+
```

Comment:
This also needs a "System.set_airplane_mode(self.target, on=False)", as done for the Jankbench workload.